src/App.js

```diff
@@ -1,21 +1,82 @@
-import React from 'react';
-import { Switch, Route } from 'react-router-dom';
+import React, { useState, useEffect } from 'react';
+import { useHistory, Switch, Route, Redirect } from 'react-router-dom';
 
-import { UserMain } from './containers/userMain/UserMain';
+import { Protected } from './components/protected/Protected';
 import { Login } from './containers/login/Login';
-import { Appointments } from './containers/appointments/Appointments';
+import { NewPatient } from './containers/newPatient/NewPatient';
+import { Provider } from './containers/provider/Provider';
+import { Patient } from './containers/patient/Patient';
 import { Navbar } from './components/navbar/Navbar';
+import { getProviders, addPatient } from './data';
 
 function App() {
+  const [loggedIn, setLoggedIn] = useState(false);
+  const [user, setUser] = useState();
+  const [navName, setNavName] = useState('Appointments App');
+  const history = useHistory();
+  const providers = getProviders();
+
+  useEffect(() => {
+    if(loggedIn){
+      history.push(`/${user.type}`);
+      const lastname = (user.lastname === '') ? '' : `${user.lastname}, `
+      setNavName(`${lastname}${user.firstname}`);
+    }
+  },[loggedIn, history, user]);
+  
+  // logs out of patient or provider page
+  const logOut = () => {
+    setLoggedIn(false);
+    setNavName('Appointments App');
```

Comment:
`'Appointments App'` should be extracted to a variable and used here and above. There should be a single source of truth.